ovs/lib/storagerouter.py

```diff
@@ -1429,38 +1438,139 @@ def get_proxy_config(vpool_guid, storagerouter_guid):
+                                                            'environment': (str, None)})
+            if stats_monkey_new_config['transport'] == 'influxdb':
+                Toolbox.verify_required_params(actual_params=stats_monkey_new_config, required_params={'username': (str, None)})
+
+        # Configure remote access
         service_manager = ServiceFactory.get_manager()
-        try:
-            for storagerouter in StorageRouterList.get_storagerouters():
-                clients.append((SSHClient(storagerouter), SSHClient(storagerouter, username='root')))
-        except UnableToConnectException:
-            raise RuntimeError('Not all StorageRouters are reachable')
-        Configuration.set('/ovs/framework/support|enabled', enable)
-        Configuration.set('/ovs/framework/support|enablesupport', enable_support)
-        for ovs_client, root_client in clients:
-            if enable_support is False:
-                root_client.run('service openvpn stop')
-                root_client.file_delete('/etc/openvpn/ovs_*')
-            if enable is True:
-                if not service_manager.has_service(StorageRouterController.SUPPORT_AGENT, client=root_client):
-                    service_manager.add_service(StorageRouterController.SUPPORT_AGENT, client=root_client)
-                service_manager.restart_service(StorageRouterController.SUPPORT_AGENT, client=root_client)
-            else:
-                if service_manager.has_service(StorageRouterController.SUPPORT_AGENT, client=root_client):
-                    service_manager.stop_service(StorageRouterController.SUPPORT_AGENT, client=root_client)
-                    service_manager.remove_service(StorageRouterController.SUPPORT_AGENT, client=root_client)
-        return True
+        if remote_access_change is True:
+            Configuration.set(key=remote_access_key, value=remote_access_new)
+            cid = Configuration.get('/ovs/framework/cluster_id').replace(r"'", r"'\''")
+            for storagerouter, root_client in root_clients.iteritems():
+                if remote_access_new is False:
+                    StorageRouterController._logger.info('Un-configuring remote access on StorageRouter {0}'.format(root_client.ip))
+                    nid = storagerouter.machine_id.replace(r"'", r"'\''")
+                    service_name = 'openvpn@ovs_{0}-{1}'.format(cid, nid)
+                    if service_manager.has_service(name=service_name, client=root_client):
+                        service_manager.stop_service(name=service_name, client=root_client)
+                    root_client.file_delete(filenames=['/etc/openvpn/ovs_*'])
+
+        # Configure support agent
+        if support_agent_change is True:
+            service_name = 'support-agent'
+            Configuration.set(key=support_agent_key, value=support_agent_new)
+            for root_client in root_clients.itervalues():
+                if support_agent_new is True:
```

Comment:
I don't like these names, but I'll let it slide

Reply:
Let it be, oh let it be
Whisper wor-ords of wisdom
...